app/Viewer.tsx

```diff
@@ -23,7 +24,65 @@ function Viewer(props: Props) {
   }, [fileInfo]);
 
   return (
-    <H5WasmProvider filename={fileInfo.name} buffer={buffer}>
+    <H5WasmProvider filename={fileInfo.name} buffer={buffer}
+        getExportURL={(format, dataset, selection, value) => {
+          if (format === 'csv') {
+            // Async function that will be called when the user clicks on a `CSV` export menu entry
+            return async () => {
+              // Generate CSV string from `value` array
+              let csv = '';
+              let dims = [];
+
+              // Record all dataset dimensions with cardinality greater than one,
+              // accounting for subset selection if applicable
+              if (selection) {
+                let subsets = selection.split(',')
+                for (let i = 0; i < subsets.length; i++) {
+                  // Because of the user interface, all slices will either be
+                  // the full array length or a single value
+                  if (subsets[i] === ':' && dataset.shape[i] > 1) {
+                    dims.push(dataset.shape[i])
+                  }
+                }
+              } else {
+                dataset.shape.forEach((val) => {
+                  if (val > 1) dims.push(val)
+                })
+              }
+
+              // Only provide special handling for 2D dataset outputs
+              if (dims.length == 2) {
+                let k = 0
+                for (let i = 0; i < dims[0]; i++) {
+                  for (let j = 0; j < dims[1] - 1; j++) {
+                    csv += `${value[k].toString()}, `
+                    k++
+                  }
+
+                  csv += `${value[k].toString()}\n`
+                  k++
+                }
+              } else {
+                value.forEach((val) => {
+                  csv += `${val.toString()}\n`
+                })
+              }
+
+              const finalCsv = csv.slice(0, -1);
```

Comment:
With my suggestion above, I believe `-1` would have to be replaced with `-2` to remove both the last line break and last coma.